ddtrace/internal/datadog/profiling/dd_wrapper/src/exporter.cpp

```diff
@@ -316,19 +326,30 @@ Profile::Profile(ProfileType type, unsigned int _max_nframes)
     values.resize(samplers.size());
     std::fill(values.begin(), values.end(), 0);
 
-    ddog_prof_Period default_sampler = { samplers[0], 1 }; // Mandated by pprof, but probably unused
-    ddog_profile = ddog_prof_Profile_new({ &samplers[0], samplers.size() }, &default_sampler, nullptr);
+    ddog_prof_Period default_period = { .type_ = samplers[0], .value = 1 }; // Mandated by pprof, but probably unused
+    ddog_prof_Slice_ValueType sample_types = { .ptr = samplers.data(), .len = samplers.size() };
+    ddog_prof_Profile_NewResult res = ddog_prof_Profile_new(sample_types, &default_period, nullptr);
+
+    // Check that the profile was created properly
+    if (res.tag != DDOG_PROF_PROFILE_NEW_RESULT_OK) {
+        std::string ddog_err(ddog_Error_message(&res.err).ptr);
+        errmsg = "Could not create profile, err: " + ddog_err;
+        ddog_Error_drop(&res.err);
+        throw std::runtime_error(errmsg);
+        return;
+    }
+    ddog_profile = res.ok;
+
+    // Initialize storage
+    locations.reserve(max_nframes + 1); // +1 for a "truncated frames" virtual frame
 
     // Prepare for use
     reset();
-
-    // Initialize the size for buffers
-    cur_frame = 0;
 }
 
 Profile::~Profile()
 {
-    ddog_prof_Profile_drop(ddog_profile);
+    ddog_prof_Profile_drop(&ddog_profile);
```

Comment:
How did this typecheck before!

Reply:
I changed the definition of `ddog_profile` in the class, so the overall type is the same.  The argument is still a pointer, but the underlying member no longer takes the value of a stack-allocated pointer.

Originally this wasn't a problem because the libdatadog3 interface returned the pointer directly (it was heap-allocated).  It also wasn't a problem in the add_echion PR (which is where I split this one from) since I read the docs and did it properly.

However, when I broke apart that PR, I took a half-measure in this PR which gave `ddog_profile` a pointer to a stack-allocated value.

The really, really curious thing is that I didn't get any segfaults from _THAT_, which is a complete mystery.  My best hypothesis is that there just so happens to be some kind of void in the stack and I got rid of it when I twiddled the interface.  Or I was testing the wrong thing.

Either way, fixed now.  🙃